app/hooks/useRedirectTo404IfFalsy.tsx

```diff
@@ -0,0 +1,13 @@
+import { useEffect } from "react";
+import { useRouter } from "next/router";
+
```

Comment:
I would add just a comment or 2 about how to use this, otherwise the semantics are a double-negative that isn't super easy to read.

```suggestion
/** 
 *  Hook that returns true if the router will redirect, false otherwise.
 *  - Example usage:
 *    const isRedirecting = useRedirectTo404IfFalsy(object);
 *    if(isRedirecting) return null;
 */
```